internal/bminventory/inventory.go

```diff
@@ -5319,12 +5346,16 @@ func (b *bareMetalInventory) updateInfraEnvData(infraEnv *common.InfraEnv, param
 	return nil
 }
 
-func (b *bareMetalInventory) validateAndUpdateInfraEnvParams(ctx context.Context, params *installer.UpdateInfraEnvParams) (installer.UpdateInfraEnvParams, error) {
+func (b *bareMetalInventory) validateAndUpdateInfraEnvParams(ctx context.Context, params *installer.UpdateInfraEnvParams, mirrorRegistryConfig *common.MirrorRegistryConfiguration) (installer.UpdateInfraEnvParams, error) {
 
 	log := logutil.FromContext(ctx, b.log)
 
 	if params.InfraEnvUpdateParams.PullSecret != "" {
-		if err := b.ValidatePullSecret(params.InfraEnvUpdateParams.PullSecret, ocm.UserNameFromContext(ctx), ""); err != nil {
+		var mirroredRegistries []string
+		if mirrorRegistryConfig != nil {
+			mirroredRegistries = extractMirroredRegistriesFromConfig(log, mirrorRegistryConfig)
```

Comment:
What happens if extraction fails?

Reply:
As agreed below let's put the comment on extractMirroredRegistriesFromConfig, that way we should only need to do it once.
I think it's reasonable for the reader to look at the function and see the comment.

Reply:
Added the comment, thanks!